gateway-server/src/main/java/org/apache/knox/gateway/services/topology/impl/DefaultTopologyService.java

```diff
@@ -790,33 +792,30 @@ private static boolean writeConfig(File dest, String name, String content) {
   /**
    * Change handler for simple descriptors
    */
-  public static class DescriptorsMonitor extends FileAlterationListenerAdaptor
-                                          implements FileFilter {
+  public static class DescriptorsMonitor extends FileAlterationListenerAdaptor implements FileFilter {
 
-    static final List<String> SUPPORTED_EXTENSIONS = new ArrayList<>();
-    static {
-      SUPPORTED_EXTENSIONS.add("json");
-      SUPPORTED_EXTENSIONS.add("yml");
-      SUPPORTED_EXTENSIONS.add("yaml");
-    }
+    static final List<String> SUPPORTED_EXTENSIONS = Arrays.asList("json", "yml", "yaml");
 
     private GatewayConfig gatewayConfig;
-
     private File topologiesDir;
-
     private AliasService aliasService;
-
+    private final TopologyService topologyService;
     private Map<String, List<String>> providerConfigReferences = new HashMap<>();
 
 
     static boolean isDescriptorFile(String filename) {
       return SUPPORTED_EXTENSIONS.contains(FilenameUtils.getExtension(filename));
     }
 
-    public DescriptorsMonitor(GatewayConfig config, File topologiesDir, AliasService aliasService) {
+    static boolean isXmlTypeDescriptorFile(String fileName) {
+      return "xml".equals(FilenameUtils.getExtension(fileName));
```

Comment:
I might even argue that the CM descriptor extension should be **cm** or something other than the generic **xml**.

Reply:
Done